packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -215,4 +217,77 @@ class Breakpoint {
 
     return isWidthActive && isHeightActive && isRightPlatform;
   }
+
+  /// Returns the currently active [Breakpoint].
+  static Breakpoint activeBreakpointOf(BuildContext context) {
+    final SlotLayout? slotLayout =
+        context.findAncestorWidgetOfExactType<SlotLayout>();
+
+    if (slotLayout != null) {
+      Breakpoint? fallbackBreakpoint;
+
+      for (final MapEntry<Breakpoint, SlotLayoutConfig?> config
+          in slotLayout.config.entries) {
+        if (config.key.isActive(context)) {
+          if (config.key.platform != null) {
+            return config.key;
+          } else {
+            fallbackBreakpoint ??= config.key;
+          }
+        }
+      }
+      if (fallbackBreakpoint != null) {
+        return fallbackBreakpoint;
+      }
+    }
+
+    final TargetPlatform platform = Theme.of(context).platform;
+    final bool isDesktop = Breakpoint.desktop.contains(platform);
+    final bool isMobile = Breakpoint.mobile.contains(platform);
+
+    for (final Breakpoint breakpoint in <Breakpoint>[
+      Breakpoints.small,
+      Breakpoints.medium,
+      Breakpoints.mediumLarge,
+      Breakpoints.large,
+      Breakpoints.extraLarge,
+    ]) {
+      if (breakpoint.isActive(context)) {
+        if (isDesktop) {
+          switch (breakpoint) {
+            case Breakpoints.small:
+              return Breakpoints.smallDesktop;
+            case Breakpoints.medium:
+              return Breakpoints.mediumDesktop;
+            case Breakpoints.mediumLarge:
+              return Breakpoints.mediumLargeDesktop;
+            case Breakpoints.large:
+              return Breakpoints.largeDesktop;
+            case Breakpoints.extraLarge:
+              return Breakpoints.extraLargeDesktop;
+            default:
+              return Breakpoints.standard;
+          }
+        } else if (isMobile) {
+          switch (breakpoint) {
+            case Breakpoints.small:
+              return Breakpoints.smallMobile;
+            case Breakpoints.medium:
+              return Breakpoints.mediumMobile;
+            case Breakpoints.mediumLarge:
+              return Breakpoints.mediumLargeMobile;
```

Comment:
Consider also breaking this down into a smaller method, which gets the breakpoint based on current plaform:

```dart
static Breakpoint getBreakpointForTargetPlatform(BuildContext context) {
  final TargetPlatform platform = Theme.of(context).platform;
  final bool isDesktop = Breakpoint.desktop.contains(platform);
  final bool isMobile = Breakpoint.mobile.contains(platform);
  
  for (final Breakpoint breakpoint in <Breakpoint>[
    Breakpoints.small,
    Breakpoints.medium,
    Breakpoints.mediumLarge,
    Breakpoints.large,
    Breakpoints.extraLarge,
  ]) {
    if (breakpoint.isActive(context)) {
      if (isDesktop) {
        switch (breakpoint) {
          case Breakpoints.small:
            return Breakpoints.smallDesktop;
          case Breakpoints.medium:
            return Breakpoints.mediumDesktop;
          case Breakpoints.mediumLarge:
            return Breakpoints.mediumLargeDesktop;
          case Breakpoints.large:
            return Breakpoints.largeDesktop;
          case Breakpoints.extraLarge:
            return Breakpoints.extraLargeDesktop;
          default:
            return Breakpoints.standard;
        }
      } else if (isMobile) {
        switch (breakpoint) {
          case Breakpoints.small:
            return Breakpoints.smallMobile;
          case Breakpoints.medium:
            return Breakpoints.mediumMobile;
          case Breakpoints.mediumLarge:
            return Breakpoints.mediumLargeMobile;
          case Breakpoints.large:
            return Breakpoints.largeMobile;
          case Breakpoints.extraLarge:
            return Breakpoints.extraLargeMobile;
          default:
            return Breakpoints.standard;
        }
      } else {
        return breakpoint;
      }
    }
  }
  return Breakpoints.standard;
}
```